### DISCUSSION TITLE
how sprite children dimensions work?

### DISCUSSION THREAD

**Discussion by k11q** [pixijs/pixijs#9009] [Q&A]
Hi I am still wrapping my mind through how dimensions work, especially for nested elements. I figured a child element x and y is relative to top left of parent, right? And how about width and height? as It seems to be different than html.

Also can I get the position of a child element relative to canvas/other elements without considering the parent?

Thank you

**Comment by miltoncandelero** [comment]
Some objects know their size because they have an intrinsic size (Sprites have a texture), others know it by the bounding box that contains all of it's children (like container).

Setting the width or the height of an object will 99.9% of the time modify the scale of the object to meet the new constraint.

(TilingSprite is the 0.1% that won't do that)

**Reply by k11q** [reply]
> Setting the width or the height of an object will 99.9% of the time modify the scale of the object to meet the new constraint.

so setting a child width or height will modify it's scale? How is it calculated? Also, what do you mean by constraint?

> 
> (TilingSprite is the 0.1% that won't do that)

coming from html i am still wrapping my head around all the types, why there couldn't be an equivalent of 'div', and modify all attributes from there?

Another sidequestion is can I have radius with sprite? What is the method to achieve that? If for example I want to replicate a 'div', can I do that?

**Reply by k11q** [reply]
If I were to replicate html/css with pixi, how may I do that? aside from basic css such as border radius, how would I leverage parent/child? I don't mind creating custom layouts, just want to find out what's already available

**Reply by miltoncandelero** [reply]
Pixijs doesn't come from html but from flash.
Containers and sprites are what once were MovieClips and Bitmaps

**Reply by k11q** [reply]
I understand that, just want to find out how the height and width is calculated
